envi/memcanvas/__init__.py

```diff
@@ -383,6 +383,40 @@ def __str__(self):
         return self.strval
 
 
+class FileBackedMemoryCanvas(StringMemoryCanvas):
```

Comment:
What is the impetus for this?

Reply:
this is a more robust way of writing the output from a VDB command to a file (including "appending".

currently, we do the overly simplistic:
* change the canvas
* do the thing
* write the file

but what if what we're trying to do doesn't complete?  what if something breaks or VDB blows up?  what if the output of the command causes OOM or some other failure (this is not hypothetical, i wrote this out of necessity)?

this approach allows us to capture and write the data pretty much as it's generated, so we don't lose context/data that we're trying to capture.